### PR TITLE
add/update agent about socket message

### PR DESCRIPTION
**Pull request by you-looks-not-tasty**
1. rearrange socket option code 
2. unify the  route of ipv4 & ipv6  socket option params
3. add ipvs/dest sockopt, update service, route, laddr, ifaddr sockopt GET & SET method

### CODE REVIEW

File: src/route.c [hunk 1]
```diff
@@ -508,138 +508,108 @@ int route_flush(void)
...
-    if (conf && size >= sizeof(*cf))
-        cf = conf;
-    else
-        cf = NULL;
-
-    if (cf && strlen(cf->ifname)) {
-        port = netif_port_get_by_name(cf->ifname);
+    char *ifname = ((struct dp_vs_route_detail*)conf)->ifname;
+    if (conf && strlen(ifname)) {
+        port = netif_port_get_by_name(ifname);
```
Comment (ywc689) [inline src/route.c:597]: `if (conf && strlen(ifname)) `:  the `conf` makes no sense in the if statements for it's already used in line:595.

File: src/route.c [hunk 2]
```diff
@@ -717,6 +687,16 @@ static int route_del_msg_cb(struct dpvs_msg *msg)
...
+static struct dpvs_sockopts agent_route_sockopts = {
+    .version        = SOCKOPT_VERSION,
+    .set_opt_min    = DPVSAGENT_ROUTE_ADD,
+    .set_opt_max    = DPVSAGENT_ROUTE_DEL,
+    .set            = route_sockopt_set,
+    .get_opt_min    = DPVSAGENT_ROUTE_GET,
+    .get_opt_max    = DPVSAGENT_ROUTE_GET,
+    .get            = route_sockopt_get,
+};
+
```
Comment (ywc689) [inline src/route.c:699]: The action functions of `agent_route_sockopts ` are the same with `route_sockopts`. Can they be combined?

File: tools/dpip/route.c [hunk 3]
```diff
@@ -177,70 +189,43 @@ static void route6_dump(const struct dp_vs_route6_conf *rt6_cfg)
...
-        } else if (strcmp(conf->argv[0], "onlink") == 0) {
-            ;/* on-link is output only */
         } else if (strcmp(conf->argv[0], "local") == 0) {
-            route->scope = ROUTE_CF_SCOPE_HOST;
-        } else if (strcmp(conf->argv[0], "table") == 0) {
-            NEXTARG_CHECK(conf, "outwall");
-            if (strcmp(conf->argv[0], "outwall") != 0)
-                return -1;
-            route->outwalltb = 1;
+            route->flags |= RTF_HOST;
```
Comment (ywc689) [inline tools/dpip/route.c:228]: Modify the help message of `dpip` for consistency.

**Comment by ywc689** [comment]
1. Open source dpvs-agent.  You can add its codes into directory `tools` of the project.
2. It's suggested to define a compile macro to enable/disable  dpvs-agent.

File: include/conf/route.h [hunk 4]
```diff
@@ -62,9 +63,21 @@ struct dp_vs_route_conf {
...
+    rt_addr_t   src;
+    rt_addr_t   gateway;
+    rt_addr_t   prefsrc;
+    char        ifname[IFNAMSIZ];
+} dpvs_route_detail_t;
+
 struct dp_vs_route_conf_array {
     int                 nroute;
-    struct dp_vs_route_conf   routes[0];
+    struct dp_vs_route_detail   routes[0];
```
[the remark this answers is not in the document]
Comment (you-looks-not-tasty) [reply include/conf/route.h:80]: The members which missing is not actually inused.  we can resume it in someday we need it.

File: src/route.c (hunk 2 above)
Comment (you-looks-not-tasty) [reply src/route.c:699]: the definition of sockopt about dpvs agent is a isolated area.  it looks more clearly

**Review by ywc689 — APPROVED** [review]

**Pull request merged**